tests/Http/Controllers/CelebTest.php

```diff
@@ -13,164 +14,334 @@ class CelebTest extends TestCase
 
     /** @test */
-    public function normal_user_cannot_access_celebs_update()
+    public function validation_produces_errors_when_admin_updates_celeb_with_all_data_being_empty()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $user = \App\Models\User::factory()->create([
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => null,
+                'date_of_birth' => null,
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'name' => 'The name field is required.',
+                'date_of_birth' => 'The date of birth field is required.',
+                'photo' => 'The photo field is required.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_date_of_birth_after_todays_date()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => 'James Islington',
+                'date_of_birth' => '08/03/2999',
+                'photo' => 'https://genericPhotos.com'
+            ])->assertSessionHasErrors([
+                'date_of_birth' => 'The date of birth must be a date before today.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_name_being_too_short()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => 'P',
+                'date_of_birth' => '08/03/2010',
+                'photo' => 'https://genericPhotos.com'
+            ])->assertSessionHasErrors([
+                'name' => 'The name must be between 2 and 30 characters.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_name_being_too_long()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => 'Beauregard Ofter-Jarms-Vendersenderson',
+                'date_of_birth' => '08/03/1989',
+                'photo' => 'https://genericPhoto.com'
+            ])->assertSessionHasErrors([
+                'name' => 'The name must be between 2 and 30 characters.'
+            ]);
+    }
+
+    /** @test */
+    public function user_cannot_update_celeb()
+    {
+        $celeb = Celeb::factory()->create();
+        $user = User::factory()->create([
             'is_admin' => false
         ]);
 
         $this->actingAs($user)
             ->putJson("celebs/$celeb->id")
-            ->assertStatus(403);
+            ->assertRedirect();
```

Comment:
For this one, what do you also want to check is if the database results did not change.

So first change the $celeb to create something with the results you want 
` $celeb = Caleb::factory()->create(['name' =>  something ])` Do that for all the fillable columns
then in the put, you should change the value of one of them. 

Finally, at the end of the test, you should assert the database has the initial values and did not change. 

This is the rule of thumb for all tests that involve changes to the DB. So DELETE and CREATE also should be like that.